drivers/sensor/st/stmemsc/stmemsc_i2c.c

```diff
@@ -21,7 +21,14 @@ int stmemsc_i2c_read(const struct i2c_dt_spec *stmemsc,
 int stmemsc_i2c_write(const struct i2c_dt_spec *stmemsc,
 		      uint8_t reg_addr, uint8_t *value, uint8_t len)
 {
-	return i2c_burst_write_dt(stmemsc, reg_addr, value, len);
+	uint8_t buf[17];
```

Comment:
Aaaargh!!
Is 16 an acceptable value to limit len? How did you come up with this value?

Reply:
it was arbitrarily chosen... I could make it a Kconfig, do you see any sensors needing to write more than 16 bytes?

Reply:
I actually don't know. It's just that I don't feel comfortable in imposing a limit there, but I understand the rationale behind it. And I'm pretty sure it is an old story, even if I failed in finding the old discussion...

What I feel would be nice is adding a macro that helps in documenting it and that can be used to divide any future attempts to write an unlimited buffer length into 16-byte burst data chunks:

```
#define STMESMC_WRITE_BUF_MAX_LEN 16
```

What do you think? Would that be useful?

Reply:
@XenuIsWatching Is it possible to change your implementation in the way suggested? What do you think?

Reply:
yes, I'll make it KConfig with a `default 16`